documentation/metrics_api.markdown

```diff
@@ -0,0 +1,267 @@
+---
+layout: default
+title: "Puppet Server: Metrics API"
+canonical: "/puppetserver/latest/metrics_api.html"
+---
+
+# The Metrics API
+
+Puppet Enterprise includes two optional, enabled-by-default web APIs for
+[Java Management Extension (JMX)](https://docs.oracle.com/javase/tutorial/jmx/index.html)
+metrics, namely
+[managed beans (MBeans)](https://docs.oracle.com/javase/tutorial/jmx/mbeans/).
+
+## Metrics v2
+
+The first uses the [Jolokia](https://jolokia.org) library and is accessible at:
+
+    /metrics/v2/
+
+Since the v2 API is implimented using a large open source metrics library with
+its own extensive documentation, below we will only walk through a very brief
+overview with the most common configuration and queries. Each section includes
+links to Jolokia's comprehensive documentation.
+
+### Configuration
+
+For security reasons only the read operations of the Jolokia interface are
+enabled by default. This includes:
+
+* `read`
+* `list`
+* `version`
+* `search`
+
+Users may change the security access policy by providing their own
+`/etc/puppetlabs/puppetserver/jolokia-access.xml` file that follows the
```

Comment:
I *think* this would only work if the user had added a `metrics-webserver.jolokia.servlet-init-params.policyLocation` setting which was set to `file:///etc/puppetlabs/puppetserver/jolokia-access.xml`.  We could set that value in the config by default but if we do that without providing our own default one in the puppetserver package, I think users would see a nasty-looking ERROR message in the puppetserver log at startup, something like:

> 2017-06-21 09:15:42,029 ERROR [p.t.s.m.jolokia] Error while accessing access restrictor at file:///etc/puppetlabs/puppetserver/jolokia-access.xml. Denying all access to MBeans for security reasons. Exception: java.io.FileNotFoundException: /etc/puppetlabs/puppetserver/jolokia-access.xml (No such file or directory)

I'm thinking it might be best to put that setting in our default config but have it be commented out by default and document here that the user would need to uncomment it, set to the desired location, and restart/reload the service.

@Sharpie - what do you think on this?